packages/uploadthing/src/internal/handler.ts

```diff
@@ -339,15 +336,32 @@ export const buildRequestHandler = <TRouter extends FileRouter>(
           });
         }
 
-        const callbackUrl = url.origin + url.pathname;
+        let callbackUrl = url;
+        if (config?.callbackUrl) {
```

Comment:
should this override `input.url`?

Reply:
input.url is a thing some of our adapters injects as some of them doesn't have an absolute URL on `req.url`. But I think the user should be able to override this as the automatic detection doesn't work everytime (as mentioned in the docs for this PR).